build.rs

```diff
@@ -119,44 +121,123 @@ fn main() {
+                    "arch/X86/X86Disassembler.c",
+                    "arch/X86/X86DisassemblerDecoder.c",
+                    "arch/X86/X86IntelInstPrinter.c",
+                    "arch/X86/X86Mapping.c",
+                    "arch/X86/X86Module.c",
+                    "arch/XCore/XCoreDisassembler.c",
+                    "arch/XCore/XCoreInstPrinter.c",
+                    "arch/XCore/XCoreMapping.c",
+                    "arch/XCore/XCoreModule.c",
+                ];
+
+                cc::Build::new()
+                    .files(files.into_iter().map(|f| format!("{}/{}", CAPSTONE_DIR, f)))
+                    .include("capstone/include")
+                    .define("CAPSTONE_X86_ATT_DISABLE_NO", None)
+                    .define("CAPSTONE_DIET_NO", None)
+                    .define("CAPSTONE_X86_REDUCE_NO", None)
+                    .define("CAPSTONE_HAS_ARM", None)
+                    .define("CAPSTONE_HAS_ARM64", None)
+                    .define("CAPSTONE_HAS_MIPS", None)
+                    .define("CAPSTONE_HAS_POWERPC", None)
+                    .define("CAPSTONE_HAS_SPARC", None)
+                    .define("CAPSTONE_HAS_SYSZ", None)
+                    .define("CAPSTONE_HAS_X86", None)
+                    .define("CAPSTONE_HAS_XCORE", None)
+                    .define("CAPSTONE_USE_SYS_DYN_MEM", None)
+                    .flag_if_supported("-Wno-unused-function")
+                    .flag_if_supported("-Wno-unused-parameter")
+                    .flag_if_supported("-Wno-unknown-pragmas")
+                    .flag_if_supported("-Wno-sign-compare")
+                    .static_crt(use_static_crt)
+                    .compile("capstone");
+            }
+
+            #[cfg(not(windows))]
+            {
+                use std::process::Command;
+                
+                // In BSDs, `make` does not refer to GNU make
+                let target_os = env_var("CARGO_CFG_TARGET_OS");                
+                let make_cmd = if target_os.contains("bsd") || target_os == "dragonfly" {
+                    "gmake"
+                } else {
+                    "make"
+                };
+
+                let out_dir = env_var("OUT_DIR");
+                Command::new(make_cmd)
+                    .current_dir(CAPSTONE_DIR)
+                    .status()
+                    .expect("Failed to build Capstone library");
+                let capstone = "libcapstone.a";
+                Command::new("cp")
+                    .current_dir(CAPSTONE_DIR)
+                    .arg(&capstone)
+                    .arg(&out_dir)
+                    .status()
+                    .expect("Failed to copy capstone library to OUT_DIR");
+
+                println!("cargo:rustc-link-search=native={}", out_dir);
+            }
         }
         header_search_paths.push(PathBuf::from("capstone/include"));
-        link_type = Some(LinkType::Static);
+        link_type = Some(LinkType::Static);        
```

Comment:
You should get rid of this trailing whitespace.

Reply:
Done!